notebook/tests/selenium/test_dualmode_insertcell.py

```diff
@@ -0,0 +1,53 @@
+
+
+def test_insert_cell(notebook):
+    a = "print('a')"
+    b = "print('b')"
+    c = "print('c')"
+
+    notebook.edit_cell(index=0, content=a)
+    notebook.append(b, c)
+    notebook.to_command_mode()
+
+    assert notebook.get_cells_contents() == [a, b, c] 
+
+    notebook.to_command_mode()
+    notebook.focus_cell(2)
+    notebook.convert_cell_type(2, "markdown")
+    
+    # insert code cell above
+    notebook.current_cell.send_keys("a")
+    assert notebook.get_cell_contents(2) == ""
+    assert notebook.get_cell_type(2) == "code"
+    assert len(notebook.cells) == 4
+    
+    # insert code cell below
+    notebook.current_cell.send_keys("b")
+    assert notebook.get_cell_contents(2) == ""
+    assert notebook.get_cell_contents(3) == ""
+    assert notebook.get_cell_type(3) == "code"
+    assert len(notebook.cells) == 5
+
+    notebook.focus_cell(2)
+    notebook.convert_cell_type(2, "markdown")
+    assert notebook.get_cell_type(2) == "markdown"
+    notebook.current_cell.send_keys("a")
+    assert notebook.get_cell_type(3) == "markdown"
+    notebook.current_cell.send_keys("b")
+    assert notebook.get_cell_type(4) == "markdown"
```

Comment:
I think the thing this is trying to test no longer works like that. We used to insert a new cell of the same type as the one you had selected, but we eventually decided it was easier to follow if it was always a code cell that was inserted. The test passes, but I think that's two accidents cancelling each other out - the cell you converted to markdown becomes number 3, then number 4, as you insert other cells.

Reply:
Oh .. okay. I will remove those steps and assertions then. Do you have any other comments?

Reply:
I was actually confused about that. I tried to do the test manually and I wasn't able to insert markdown cells. It all makes sense now

Reply:
Also, I deeply apologize for the delay in submitting this PR. I will do my best to make sure this doesn't happen again and I hope this doesn't affect your idea of how committed I am to the team and the project.

Reply:
No need to apologise! Everyone gets busy at times, and no-one expects volunteer contributors to do things by any set deadline. We might ping you on unfinished pull requests in case you've forgotten about it, or in case you're waiting for us to answer something, but if you don't have time to work on it, that's fine. If it's urgent, someone else can pick it up and finish it, and if not, it can wait. :-)